### ISSUE TITLE
Edition <2021 `uninlined_format_args` fix causes `non_fmt_panics`

### ISSUE THREAD

**Issue by alexheretic** [rust-lang/rust-clippy#10236]
### Summary

`uninlined_format_args` can be false positive for older edition, e.g. 2018, where some macros do not support such formatting.

### Lint Name

uninlined_format_args

### Reproducer

I tried this code:

```rust
// Cargo.toml edition = "2018"
let v = vec![1];
assert!(v.is_empty(), "v non-empty: {:?}", v);
```

I saw this happen:

```
warning: variables can be used directly in the `format!` string
  --> src/main.rs:11:5
   |
11 |     assert!(v.is_empty(), "v non-empty: {:?}", v);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args
   = note: `#[warn(clippy::uninlined_format_args)]` on by default
help: change this to
   |
11 -     assert!(v.is_empty(), "v non-empty: {:?}", v);
11 +     assert!(v.is_empty(), "v non-empty: {v:?}");
   |
```

I expected to see this happen: 

No such warning unless on at least edition 2021.

If the fix is applied to older editions it'll result in the rustc warning
```
warning: panic message contains an unused formatting placeholder
 --> src/main.rs:9:37
  |
9 | assert!(v.is_empty(), "v non-empty: {v:?}");
  |                                     ^^^^^
  |
  = note: this message is not used as a format string when given without arguments, but will be in Rust 2021
```

And indeed it will not work outputting `thread 'main' panicked at 'v non-empty: {v:?}'` (instead of `thread 'main' panicked at 'v non-empty: [1]'`)

### Version

```text
rustc 1.67.0 (fc594f156 2023-01-24)
binary: rustc
commit-hash: fc594f15669680fa70d255faec3ca3fb507c3405
commit-date: 2023-01-24
host: x86_64-unknown-linux-gnu
release: 1.67.0
LLVM version: 15.0.6
```

**Comment by taiki-e** [comment]
This was fixed in https://github.com/rust-lang/rust/commit/e5010c996e9cb9fa09fefb4e3d31608915869c78 (https://github.com/rust-lang/rust-clippy/pull/10055), but that patch is only included in 1.68 and later, not 1.67.

**Comment by taiki-e** [comment]
(Also this issue is duplicate of #10234)

**Comment by alexheretic** [comment]
thanks
